backend/service/sourcegraph/sourcegraph.go

```diff
@@ -88,14 +88,16 @@ func (c *client) CompareCommits(ctx context.Context, req *sourcegraphv1.CompareC
 		"head": graphql.String(req.Head),
 	}
 
-	err := c.gqlClient.Query(ctx, &compareCommitsQuery, variables)
+	query := &compareCommitsQuery
```

Comment:
I think this was the primary mistake, not creating a reference for the current query.

Below when we check the nodes I suppose it could of been ant query in flight which had the mismatch of length